cupyx/scipy/sparse/linalg/_interface.py

```diff
@@ -0,0 +1,679 @@
+import warnings
+
+import cupy as cp
+
+from cupyx.scipy.sparse import isspmatrix
+from cupyx.scipy.sparse._util import isshape, isintlike
+
+__all__ = ['LinearOperator', 'aslinearoperator']
+
+
+class LinearOperator(object):
+    """Common interface for performing matrix vector products
+
+    Args:
+        shape (tuple):  Matrix dimensions ``(M, N)``.
+        matvec (callable f(v)):  Returns returns ``A * v``.
+        rmatvec (callable f(v)):  Returns ``A^H * v``, where ``A^H`` is the
+                                  conjugate transpose of ``A``.
+        matmat (callable f(V)):  Returns ``A * V``, where ``V`` is a dense
+                                 matrix with dimensions ``(N, K)``.
+        dtype (dtype):  Data type of the matrix.
+        rmatmat (callable f(V)):  Returns ``A^H * V``, where ``V`` is a dense
+                                  matrix with dimensions ``(M, K)``.
+        args (tuple):  For linear operators describing products etc. of other
+                        linear operators, the operands of the binary operation.
+        ndim (int):  Number of dimensions (this is always 2)
+
+    .. seealso:: :class:`scipy.sparse.LinearOperator`
+    """
+
+    ndim = 2
+
+    def __new__(cls, *args, **kwargs):
+        if cls is LinearOperator:
+            # Operate as _CustomLinearOperator factory.
+            return super(LinearOperator, cls).__new__(_CustomLinearOperator)
+        else:
+            obj = super(LinearOperator, cls).__new__(cls)
+
+            if (type(obj)._matvec == LinearOperator._matvec
+                    and type(obj)._matmat == LinearOperator._matmat):
+                warnings.warn('LinearOperator subclass should implement'
+                              ' at least one of _matvec and _matmat.',
+                              category=RuntimeWarning, stacklevel=2)
+
+            return obj
+
+    def __init__(self, dtype, shape):
+        """Initialize this LinearOperator.
+
+        To be called by subclasses. ``dtype`` may be None; ``shape`` should
+        be convertible to a length-2 tuple.
+        """
```

Comment:
Please fix docstrings in the same way.
```suggestion
        """Initialize this :class:`LinearOperator`.
        """
```

Reply:
sorry about that ✌🏾 😅